sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/JsonSerializable.java

```diff
@@ -69,29 +69,54 @@ protected JsonSerializable(String jsonString) {
         this.propertyBag = objectNode;
     }
 
-    private ObjectMapper getMapper() {
-        // TODO: Made package private due to #153. #171 adding custom serialization options back.
-        if (this.om != null) { return this.om; }
-        return OBJECT_MAPPER;
-    }
-
-    void setMapper(ObjectMapper om) {
-        this.om = om;
-    }
-
     private static void checkForValidPOJO(Class<?> c) {
         if (c.isAnonymousClass() || c.isLocalClass()) {
             throw new IllegalArgumentException(
-                    String.format("%s can't be an anonymous or local class.", c.getName()));
+                String.format("%s can't be an anonymous or local class.", c.getName()));
         }
         if (c.isMemberClass() && !Modifier.isStatic(c.getModifiers())) {
             throw new IllegalArgumentException(
-                    String.format("%s must be static if it's a member class.", c.getName()));
+                String.format("%s must be static if it's a member class.", c.getName()));
         }
     }
 
+    static Object getValue(JsonNode value) {
+        if (value.isValueNode()) {
+            switch (value.getNodeType()) {
+                case BOOLEAN:
+                    return value.asBoolean();
+                case NUMBER:
+                    if (value.isInt()) {
+                        return value.asInt();
+                    } else if (value.isLong()) {
+                        return value.asLong();
+                    } else if (value.isDouble()) {
+                        return value.asDouble();
+                    }
+                    break;
```

Comment:
instead of `break`, can we return `value` here? I think it will make things more clear. 
We can add a `else` block at the end, and just return `value` from else condition.

Reply:
Done